packages/devextreme/js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -577,11 +586,41 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
         };
     },
 
-    _initMarkup: function() {
+    _initMarkup() {
         this._renderScrollableContainer();
         this._renderEmptyMessage(this._dataAdapter.getRootNodes());
+
         this.callBase();
-        this.setAria('role', 'tree');
+
+        this._setAriaRoleToElement();
+        this._setTabIndexToFirstNodeContainer();
+    },
+
+    _setTabIndexToFirstNodeContainer() {
+        const $scrollableContainer = this.$element().find(`.${SCROLLABLE_CONTAINER_CLASS}`);
+        const $scrollableContent = this.$element().find(`.${SCROLLABLE_CONTENT_CLASS}`);
+
+        const scrollableContentHeight = getHeight($scrollableContent);
+        const scrollableContainerHeight = getHeight($scrollableContainer);
+
+        const isContentHeightExceeded = scrollableContentHeight > scrollableContainerHeight;
+
+        if(!isContentHeightExceeded) {
+            return;
+        }
+
+        const { tabIndex } = this.option();
+        const $nodeContainer = this.$element().find(`.${NODE_CONTAINER_CLASS}`).first();
+
+        $nodeContainer.attr({ tabindex: tabIndex });
```

Comment:
It's really strange for me to see that adding of a tabIndex depends on a calculated height of a nested scrollable content.
There should be another more simple and common solution.

Reply:
We decided to remove tabindex setting in this case.